registries/adapters/local_openshift_adapter.go

```diff
@@ -69,87 +71,55 @@ func (r LocalOpenShiftAdapter) GetImageNames() ([]string, error) {
 		return nil, err
 	}
 
-	images, err := openshiftClient.ListRegistryImages()
-	if err != nil {
-		log.Errorf("Failed to load registry images")
-		return nil, err
+	imageClient := openshiftClient.Image()
+	if r.Config.Namespaces == nil {
+		log.Debug("Didn't find any namespace in broker configuration, assuming `openshift`.")
+		r.Config.Namespaces = append(r.Config.Namespaces, "openshift")
 	}
-
 	imageList := []string{}
-	for _, image := range images.Items {
-		imageList = append(imageList, strings.Split(image.DockerImageManifest, "@")[0])
+	for _, ns := range r.Config.Namespaces {
+		is, err := imageClient.ImageStreams(ns).List(meta_v1.ListOptions{})
+		if err != nil {
+			log.Errorf("Failed to get list of imagestreams for namespace [%v]: %v", ns, err)
+			continue
+		}
+		for _, i := range is.Items {
+			imageList = append(imageList, fmt.Sprintf("%v/%v", ns, i.Name))
```

Comment:
Curious about implications (are there any?) of using `%v` in places where we expect a string. Do you get anything valuable out of using `%s` instead? I don't know the answer.

Reply:
@djwhatle %v is almost always okay to use regardless of the type.

```
%v	the value in a default format
	when printing structs, the plus flag (%+v) adds field names
```